packages/blade/src/components/Checkbox/_decisions/decisions.md

```diff
@@ -0,0 +1,263 @@
+      {children}
+      <CheckboxGroupHintText variant="help | error" />
+    </CheckboxGroupFieldset>
+  );
+};
+```
+
+## `Checkbox` API
+
+| Prop                | Type       | Required | Default   | Description                                                                                                                                              |
+| ------------------- | ---------- | -------- | --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| children            | `string`   | No       | undefined | The text to be rendered as CheckboxLabel                                                                                                                 |
+| helpText          | `string`   | No       | undefined | The helper text to be rendered                                                                                                                           |
+| isChecked           | `boolean`  | No       | false     | state of the checkbox, if explicitly set the checkbox will be controlled                                                                                 |
+| defaultChecked      | `boolean`  | No       | false     | default state of the checkbox, if isChecked is not provided the checkbox will be uncontrolled                                                            |
+| onChange            | `Function` | No       | undefined | The function to be called when the checkbox state changes.                                                                                               |
+| isIndeterminate     | `boolean`  | No       | false     | mixed state of the checkbox                                                                                                                              |
+| name (html native)  | `string`   | No       | undefined | The name of the input field in a checkbox, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)            |
+| value (html native) | `string`   | No       | undefined | The value of the input field in a checkbox, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#value) |
+| isDisabled          | `boolean`  | No       | `false`   | Control whether the checkbox is disabled or not.                                                                                                         |
+| hasError           | `boolean`  | No       | `false`   | Control whether the checkbox is invalid or not.                                                                                                          |
+| isRequired          | `boolean`  | No       | `false`   | Control whether the checkbox is required or not.                                                                                                         |
```

Comment:
Didn't we conclude that we'll remove this?

Reply:
Hmm.. missed this part but we concluded to remove it on the CheckboxGroup in addition of showOptionalLabel, 
The user can mark individual Checkbox isRequired right if not used with CheckboxGroup?

Reply:
I think we decided to remove `isRequired` from everywhere and just have `showOptionalLabel` on the CheckboxGroup. All the validations for if a checkbox has to be checked or not will be dependent on the logic provided by consumers.

Reply:
Oh 🤔 

We also discussed along the lines of "The checkbox will not be used outside of CheckboxGroup" something like that right? If so then it's fine to remove isRequired. 

But if say the checkbox can be used outside of <CheckboxGroup then we should consider isRequired. for html forms and screen readers.

Reply:
@chaitanyadeorukhkar @nashcheez Let's keep isRequired. since now we have a better API `neccessityIndicator=required|optional` we can propagate it to the checkboxes to have better screen reader accessibility and set the native `required` prop.